### DISCUSSION TITLE
Use the parser for playstation

### DISCUSSION THREAD

**Discussion by KhloyannareK** [CLIUtils/CLI11#810] [Ideas]
I used this parser and wrote a parser wrapper and a parameter manager to keep the values of the options and tried it out on playstation examples. But seems this lib is not portable for compiling for PS consoles.
Shouldn’t the parser be cross-platform ?

**Comment by phlptp** [comment]
We use it on a number of different platforms, so it is cross platform.  Can you post the specific issues encountered on the PS console?    I don't think it is one we have ever attempted.

**Reply by KhloyannareK** [reply]
Hi Philip,
I try to build PS4(SDK 9.5) example ("trophy" example)  on visual studio and the following errors are emitted from the parser header file.
```
E0304    no instance of overloaded function "CLI::detail::value_string" matches the argument list    Line 2498
E0304    no instance of overloaded function "CLI::detail::value_string" matches the argument list    Line 2501
E0308    more than one instance of overloaded function "CLI::detail::checked_multiply" matches the argument Line 4703
E0135    namespace "std" has no member "getenv" Line 7972
E0308    more than one instance of overloaded function "CLI::detail::checked_multiply" matches the argument list:   Line 8690 
Error        no member named 'getenv' in namespace 'std'  Line 7972
```

I commented out the line for getenv call and the build passes but the E0*** errors from visual studio still exist.

Thanks
